src/net/parse.h

```diff
@@ -67,3 +71,118 @@ namespace net
         get_ipv4_subnet_address(boost::string_ref address, bool allow_implicit_32 = false);
 }
 
+#if defined(SEKRETA)
+namespace net::sekreta
+{
+//! \brief CLI system argument parser for Sekreta
+//!
+//! \details Offers extensibility and future-proofing for arg-caller
+//!   implementations
+//!
+//! \tparam t_address Returned daemon address type
+//! \tparam t_value Interface value type
+//!
+//! \param daemon Pair: daemon host address and port
+//! \param systems Requested system type to parse
+//!
+//! \note Currently limited to parsing for a single-system implementation and
+//!   for a single address
+//!
+//! \note Sekreta's impl helper design can serve as a framework-replacement for
+//!   CN copypasta
+//!
+//! \todo epee + serialization needs std::string_view support
+//!
+//! \todo network address parsing (any anon network parsing) should be used
+//!   through Sekreta when implemented
+template <
+    typename t_address = expect<epee::net_utils::network_address>,
+    typename t_value = std::string>
+std::pair<std::optional<t_address>, std::optional<t_value>> parse_cli_arg(
+    const std::pair<std::string, uint16_t>& daemon,
+    const std::vector<t_value>& systems)
+{
+  namespace impl = ::sekreta::api::impl_helper;
+  namespace type = impl::type;
+
+  using t_args = impl::Args<t_value>;
+
+  auto const cli_error = [&](const type::kError error) {
+    return std::make_pair(
+        std::nullopt, impl::args_error<t_args, t_value>(error).second);
+  };
+
+  bool is_garlic{false}, is_onion{false}, is_loki{false};
+  for (const auto& system : systems)
+    {
+      std::optional<type::kSystem> const arg =
+          t_args::template get_key<type::kSystem>(system);
+
+      if (!arg)
+        return cli_error(type::kError::System);
+
+      // TODO(unassigned): see below regarding SEK-key for more networks
+      switch (arg.value())
+        {
+          case type::kSystem::Kovri:
+            //case type::kSystem::Ire:
+            //case type::kSystem::I2P:
+            is_garlic = true;
+            break;
+          //case type::kSystem::Tor:
+          //  is_onion = true;
+          //  break;
+          //case type::kSystem::Loki:
```

Comment:
Anonimal I will like to see the LokiNet supported. I have learned that the Loki software is a competitor to the Monero
LGTM